### PR TITLE
[Docs] Add sharding/split-groups to Compactor capacity planning

### LINKED ISSUE #4620: Docs: Compactor Planning Capacity page should include shard/split-groups sizing
`-compactor.split-groups` `-compactor.split-and-merge-shards` are two very important settings for working with large tenants, we should include them in the `Planning capacity` page.

### PR DESCRIPTION
**Pull request by eamonryan**
#### What this PR does

Adds documentation to the `Planning Capacity` page to include how to size `-compactor.split-groups` and
`-compactor.split-and-merge-shards`.

#### Which issue(s) this PR fixes or relates to

Fixes #4620 

#### Checklist

- [ ] Tests updated
- [X] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/sources/mimir/operators-guide/run-production-environment/planning-capacity.md [hunk 1]
```diff
@@ -154,6 +154,20 @@ For more information about disk requirements, refer to [Compactor disk utilizati
...
 
+There are two compactor specific settings related to working with large tenants that should be considered when planning/adjusting capacity.
+
+These are:
+`-compactor.split-groups`
+`-compactor.split-and-merge-shards`
+
+For more information on how they work, refer to [Compaction algorithm]({{< relref "../architecture/components/compactor/index.md#compaction-algorithm" >}}).
+
+We currently recommend 1 shard per every 8 million active series in a tenant.
```
Comment (pstibrany) [inline docs/sources/mimir/operators-guide/run-production-environment/planning-capacity.md:165]: Latest recommendation I'm aware of was 25M series for a tenant. We were discussing 8M internally, but we haven't switched to that yet.

File: docs/sources/mimir/operators-guide/run-production-environment/planning-capacity.md [hunk 2]
```diff
@@ -154,6 +154,20 @@ For more information about disk requirements, refer to [Compactor disk utilizati
...
+
+For example, for a tenant with 100 million active series, use approximately 12 shards. The shard count should be an even number.
+
+We also recommend setting the number of split-groups to be the same as the shard count.
```
Comment (pstibrany) [inline docs/sources/mimir/operators-guide/run-production-environment/planning-capacity.md:169]: Split-groups is a function of number of ingesters that single tenant uses. As such it depends on replication factor. Ideally we would be splitting such that each group has not more than ~20 blocks.

I think formula should be `groups = (series × replication_factor / series_per_ingester) / blocks_per_group`.

For 100M series, 3x replication and 1.5M series per ingesters, if we want to target 20 blocks per group that would be 10 groups. (With 12 groups we get ~17 blocks per group, that's not too bad)

File: docs/sources/mimir/operators-guide/run-production-environment/planning-capacity.md [hunk 3]
```diff
@@ -154,6 +154,20 @@ For more information about disk requirements, refer to [Compactor disk utilizati
 sum(prometheus_tsdb_head_series)
 ```
 
+There are two compactor specific settings related to working with large tenants that should be considered when planning/adjusting capacity.
+
+These are:
+`-compactor.split-groups`
+`-compactor.split-and-merge-shards`
+
+For more information on how they work, refer to [Compaction algorithm]({{< relref "../architecture/components/compactor/index.md#compaction-algorithm" >}}).
```
Comment (osg-grafana) [inline docs/sources/mimir/operators-guide/run-production-environment/planning-capacity.md:163]:
```suggestion
For more information about how they work, refer to [Compaction algorithm]({{< relref "../architecture/components/compactor/index.md#compaction-algorithm" >}}).
```

File: docs/sources/mimir/operators-guide/run-production-environment/planning-capacity.md [hunk 4]
```diff
@@ -154,6 +154,20 @@ For more information about disk requirements, refer to [Compactor disk utilizati
...
+
+We currently recommend 1 shard per every 8 million active series in a tenant.
+
+For example, for a tenant with 100 million active series, use approximately 12 shards. The shard count should be an even number.
```
Comment (osg-grafana) [inline docs/sources/mimir/operators-guide/run-production-environment/planning-capacity.md:167]:
```suggestion
For example, for a tenant with 100 million active series, use approximately 12 shards. Use an even number for the shard count.
```

**Review by osg-grafana — APPROVED** [review]
Unblocking with nits, and with the assumption that an engineer further approves the technical accuracy

**Review by pracucci — COMMENTED** [review]
I'm wondering if we really should document it in the capacity planning doc page. The capacity planning doc page purpose was:

> The information that follows is an overview about the CPU, memory, and disk space that Grafana Mimir requires at scale.

The split-and-merge compactor config doesn't look much related to capacity planning to me, but more to "how to scale mimir".  What's about moving the content of your PR to `docs/sources/mimir/operators-guide/run-production-environment/production-tips/index.md`?

**Comment by eamonryan** [comment]
> The information that follows is an overview about the CPU, memory, and disk space that Grafana Mimir requires at scale.

I do see that, however, if you take a potential user who is looking to migrate to Mimir from their existing solution and they already operate at some specific high scale - this is the doc they will read since it helps them estimate resources and hopefully tells them anything else they need to pay attention to at a specific size.

Is this setting related to "capacity planning"? I think it depends on the viewpoint .

Will your installation struggle under certain conditions if you do not pay attention to this setting? It will (large tenant setup).

I don't necessarily think it's impossible to put the setting elsewhere though, but overall I think the options are:

- Put it here and anywhere that is relevant 
- Link to the appropriate other page where it is mentioned from this page, such as Production Tips, as suggested (but is it just a "tip" if your installation will be heavily affected if this setting is not correct?)
- Find some way for the setting to be auto-configured correctly for a given installation (much harder)

Happy to do whatever makes sense though.

**Comment by pracucci** [comment]
> Will your installation struggle under certain conditions if you do not pay attention to this setting? It will (large tenant setup).

The settings you're talking about go way beyond the compactors. I 💯 with you the need of a doc page for that, which is what we attempted with the "production tips" doc page. I'm wondering if we could just link it from the capacity planning doc, and keep evolving the "production tips" (maybe even restructuring and renaming it).

**Comment by lamida** [comment]
The CHANGELOG has just been cut to prepare for the next Mimir release. Please rebase `main` and eventually move the CHANGELOG entry added / updated in this PR to the top of the CHANGELOG document. Thanks!

**Comment by pracucci** [comment]
@eamonryan Have you considered this?

https://github.com/grafana/mimir/pull/4621#issuecomment-1504783389

**Comment by osg-grafana** [comment]
@cristiangsp, would you mind taking a look at this issue to gain technical context?

**Comment by GrafanaWriter** [comment]
@tacole02 - Would you mind checking this old PR for validity and either merging or closing?

**Comment by tacole02** [comment]
Hello @eamonryan ! I'm just reviewing this very old PR. Would you mind letting me know if this is still relevant/accurate/something we should merge? Thanks! 😄

**Comment by eamonryan** [comment]
Will get to this tomorrow (hoping) and review new PR w/ Taylor.

**Comment by tacole02** [comment]
Thanks so much, @eamonryan . Feel free to ping me with the new PR.

**Comment by eamonryan** [comment]
We are going to close this one and revisit it in another PR (myself and @tacole02)

**Pull request closed without merging**